dart/lib/src/sentry_tracer.dart

```diff
@@ -109,18 +109,23 @@ class SentryTracer extends ISentrySpan {
       }
 
       var _rootEndTimestamp = commonEndTimestamp;
+
+      // Trim the end timestamp of the transaction to the very last timestamp of child spans
       if (_trimEnd && children.isNotEmpty) {
-        final childEndTimestamps = children
-            .where((child) => child.endTimestamp != null)
-            .map((child) => child.endTimestamp!);
-
-        if (childEndTimestamps.isNotEmpty) {
-          final oldestChildEndTimestamp =
-              childEndTimestamps.reduce((a, b) => a.isAfter(b) ? a : b);
-          if (_rootEndTimestamp.isAfter(oldestChildEndTimestamp)) {
-            _rootEndTimestamp = oldestChildEndTimestamp;
+        DateTime? latestEndTime;
+
+        for (var child in children) {
+          final childEndTimestamp = child.endTimestamp;
+          if (childEndTimestamp != null) {
+            if (latestEndTime == null || childEndTimestamp.isAfter(latestEndTime)) {
+              latestEndTime = child.endTimestamp;
+            }
```

Comment:
`l`: You don't use `childEndTimestamp` here ` latestEndTime = child.endTimestamp`. I think you can ditch the extra variable or use it everywhere
```suggestion
          if (child.endTimestamp != null) {
            if (latestEndTime == null || child.endTimestamp.isAfter(latestEndTime)) {
              latestEndTime = child.endTimestamp;
            }
          }
        }
```

Reply:
this is a dart type promotion issue:

```dart
```

Reply:
this is a dart type promotion issue:

```dart
if (child.endTimestamp != null) {
 // child.endTimestamp is still handled as nullable because type promotion for these cases are only supported in higher dart versions
}

// vs

final childEndTimestamp = child.endTimestamp;
if (childEndTimestamp != null) {
  // childEndTimestamp is not nullable
}
  

```
declaring the variable first explcitlity is a small workaround for that issue.